### PR TITLE
Adding fileUpload feature to the CKEditor

### LINKED ISSUE #757: CKEditor - Add Enhanced Image Plugin
@jxmai looks like this has been a popular plugin request i wonder if we should just add it to the CKEditor by default?

Enhanced Image Plugin = http://ckeditor.com/addon/image2

**Requests:**
http://forum.primefaces.org/viewtopic.php?f=14&t=48571

http://forum.primefaces.org/viewtopic.php?f=14&t=48585

### PR DESCRIPTION
**Pull request by krishnakittu**

### CODE REVIEW

File: core/src/main/resources/META-INF/resources/primefaces-extensions/ckeditor/1-ckeditor-widget.js [hunk 1]
```diff
@@ -132,6 +135,7 @@ PrimeFaces.widget.ExtCKEditor = PrimeFaces.widget.DeferredWidget.extend({
                 // Issue #779: enter/shift enter Mode
                 thisConfig.enterMode = PrimeFaces.csp.evalResult(this.cfg.enterMode);
                 thisConfig.shiftEnterMode = PrimeFaces.csp.evalResult(this.cfg.shiftEnterMode);
+				thisConfig.filebrowserUploadUrl = this.cfg.fileBrowserUploadUrl;
                 // Issue #151: font and font size
```
Comment (melloware) [inline core/src/main/resources/META-INF/resources/primefaces-extensions/ckeditor/1-ckeditor-widget.js:139]: Looks like some indenting problems here. Check Tabs vs Spaces.

File: core/src/main/resources/META-INF/resources/primefaces-extensions/ckeditor/1-ckeditor-widget.js [hunk 2]
```diff
@@ -65,6 +65,9 @@ PrimeFaces.widget.ExtCKEditor = PrimeFaces.widget.DeferredWidget.extend({
         if (this.cfg.customConfig) {
             this.options.customConfig = this.cfg.customConfig + "?resolve=false";
         }
+		if (this.cfg.fileBrowserUploadUrl) {
```
Comment (melloware) [inline core/src/main/resources/META-INF/resources/primefaces-extensions/ckeditor/1-ckeditor-widget.js]: Same here check tabs vs spaces.

**Review by melloware — CHANGES_REQUESTED** [review]

**Review by melloware — APPROVED** [review]

File: core/src/main/resources/META-INF/primefaces-extensions.taglib.xml [hunk 3]
```diff
@@ -2695,6 +2695,15 @@
             <required>false</required>
             <type>java.lang.String</type>
         </attribute>
+        <attribute>
+            <description>
+                <![CDATA[The fileBrowserUploadUrl  attribute specifies fileUpload POST Url.]]>
```
Comment (jepsar) [inline core/src/main/resources/META-INF/primefaces-extensions.taglib.xml:2700]: I'd say "Specifies the file upload POST URL."

Comment (melloware) [reply core/src/main/resources/META-INF/primefaces-extensions.taglib.xml:2700]: I will fix

**Comment by krishnakittu** [comment]
Are we need to add this to the release changes?

**Comment by melloware** [comment]
i fixed the wording in the taglib and committed

**Pull request merged**